api/test/test_play_asset_pack_manager.gd

```diff
@@ -163,6 +163,267 @@ func test_cancel_asset_pack_request_non_existing_pack_name():
 	var test_result : bool = test_object.cancel_asset_pack_request(test_pack_name)
 	assert_eq(test_result, false)
 
+func test_fetch_asset_pack_success():
+	var mock_plugin = FakeAndroidPlugin.new()
+
+	# configure what should be emitted upon fetch_asset_pack() call
+	var test_pack_name = "testPack"
+	
+	var test_asset_pack_state = create_mock_asset_pack_state_with_status_and_progress_dict(test_pack_name, \
+		PlayAssetPackManager.AssetPackStatus.PENDING, 0, 4096)
+	
+	# the plugin call will return an AssetPackStates dictionary enclosing the given test_asset_pack_state
+	var test_asset_pack_states = create_mock_asset_pack_states_with_single_state_dict(test_asset_pack_state)
+	var signal_info = FakePackStatesInfo.new(true, \
+		test_asset_pack_states, {})
+	mock_plugin.set_fake_fetch_info(signal_info)
+	
+	var test_object = create_play_asset_pack_manager(mock_plugin)
+	
+	var request_object = test_object.fetch_asset_pack(test_pack_name)
+	# weakref used to test for memory leak
+	var request_object_reference = weakref(request_object)
+	
+	# connect to helper function, simulating the workflow of connecting callback to signal
+	request_object.connect("request_completed", self, "assert_fetch_signal_is_success")
+	
+	# yield to the request_completed signal for no longer than 1 seconds and assert for signal emitted
+	yield(yield_to(request_object, "request_completed", 1), YIELD)
```

Comment:
Ideally there also be a ```state_updated``` signal with the same result field, emitted around the same time as that of this ```request_completed```. 
However, since Gut can not yield to 2 signals at the same time, I decided to simplify this test logic to only yielding and asserting on ```request_completed```.